transaction-coordinator/src/main/java/org/apache/kafka/coordinator/transaction/TransactionCoordinatorRecordSerde.java

```diff
@@ -16,32 +16,29 @@
  */
 package org.apache.kafka.coordinator.transaction;
 
+import org.apache.kafka.common.errors.UnsupportedVersionException;
 import org.apache.kafka.common.protocol.ApiMessage;
 import org.apache.kafka.coordinator.common.runtime.CoordinatorLoader;
 import org.apache.kafka.coordinator.common.runtime.CoordinatorRecordSerde;
-import org.apache.kafka.coordinator.transaction.generated.TransactionLogKey;
-import org.apache.kafka.coordinator.transaction.generated.TransactionLogValue;
+import org.apache.kafka.coordinator.transaction.generated.CoordinatorRecordType;
 
 public class TransactionCoordinatorRecordSerde extends CoordinatorRecordSerde {
 
     @Override
     protected ApiMessage apiMessageKeyFor(short recordType) {
-        switch (recordType) {
-            case 0:
-                return new TransactionLogKey();
-            default:
-                throw new CoordinatorLoader.UnknownRecordTypeException(recordType);
+        try {
+            return CoordinatorRecordType.fromId(recordType).newRecordKey();
+        } catch (UnsupportedVersionException ex) {
+            throw new CoordinatorLoader.UnknownRecordTypeException(recordType);
         }
     }
 
     @Override
     protected ApiMessage apiMessageValueFor(short recordVersion) {
```

Comment:
This really is the `recordType` again I think here. The version of the `TransactionLogValue` will have no bearing on the object which is created. It needs to be 0 here whether it's a v0 or v1 value.

Reply:
Again :). Fixed it.